### ISSUE TITLE
feat: widen trainer.* RPC for live progress + full RLSL + RunManager run lifecycle

### ISSUE THREAD

**Issue by dahai80** [dahai80/fusion-agent-studio#166]
## Background

fusion-trainer landed a `RunManager` API (v0.1.9+, `fusion_trainer.run_manager.RunManager`) that gives fusion-studio's GUI everything it needs: async run start, **live progress event streaming**, run registry/history, presets/datasets/adapters listing, and run stop. Contract + high-fidelity GUI mockups: `fusion-trainer/architecture/fusion-trainer-gui.md` §2.2/§2.3.

The current `trainer.*` RPC surface in `agent_runtime/dispatchers/trainer.py` + `agent_runtime/training_service.py` is thin and has gaps the GUI needs filled:

| Gap | Current | Needed |
|---|---|---|
| **live progress** | `trainer.runs.status` = terminal status only, no event stream | `trainer.runs.progress {run_id, since_step} → {events:[...]}` for live loss curves |
| **method=rlsl** | `run_rlsl` validates `method ∈ dpo\|orpo\|grpo` — **missing rlsl** (full RM→GRPO closed loop) | allow `method=rlsl` |
| **full TrainerConfig** | `run_sft`/`run_rlsl` take loose `{model, trace_ids, preset, dataset}` — preset+flags only | accept full `TrainerConfig` JSON so the GUI's advanced fields reach the trainer |
| **run history** | `_RUNS` in-memory dict, lost on restart | `trainer.runs.list {limit} → [...]` backed by RunManager disk registry |
| **presets/datasets/adapters** | only `trainer.info` | `trainer.presets.list`, `trainer.datasets.list`, `trainer.datasets.preview`, `trainer.adapters.list`, `trainer.adapters.delete` |
| **stop** | none | `trainer.runs.stop {run_id} → {status:"cancelling"}` |

## Proposed RPC table (extends existing 5 methods)

| method | params | returns |
|---|---|---|
| `trainer.info` | `{}` | `{version, mlx_port, mlx_alive, models[], gpu_mem_gb, presets[], warnings[]}` |
| `trainer.presets.list` | `{kind:"sft"\|"rlsl"}` | `[{name,kind,description,memory_estimate_gb,hyperparams}]` |
| `trainer.datasets.list` | `{}` | `[{name,category,n_samples,format,path}]` |
| `trainer.datasets.preview` | `{name,limit}` | `{rows[],fields[],n_total}` |
| `trainer.sft.start` | `TrainerConfig(JSON)` | `{run_id,status}` |
| `trainer.rlsl.start` | `TrainerConfig(JSON, method∈dpo\|orpo\|grpo\|rlsl)` | `{run_id,status}` |
| `trainer.runs.list` | `{limit}` | `[{run_id,name,method,status,created,final_adapter,adapter_name,error}]` |
| `trainer.runs.status` | `{run_id}` | `{status,step,total_steps,metrics,phase,job_id}` |
| `trainer.runs.progress` | `{run_id, since_step}` | `{events:[{type,step,...}]}` |
| `trainer.runs.stop` | `{run_id}` | `{status:"cancelling"}` |
| `trainer.adapters.list` | `{model?}` | `[{name,base_model,method,adapter_path,run_id}]` |
| `trainer.adapters.delete` | `{name}` | `{ok}` |

## Live progress event schema (§2.3)

RunManager writes per-run `events.jsonl`; `trainer.runs.progress` replays events with `step > since_step`:

```jsonl
{"type":"train_loss","step":120,"train_loss":0.4123,"learning_rate":1.8e-05,"tokens_per_second":820,"it_sec":2.1,"peak_memory_gb":15.8}
{"type":"val_loss","step":120,"val_loss":0.3891}
{"type":"grpo_step","step":1,"loss":0.21,"mean_reward":0.62,"mean_ratio":1.04,"n_samples":8}
{"type":"phase","phase":1,"name":"reward_model_train"}
{"type":"status","status":"completed","adapter_path":"...","adapter_name":"..."}
```

## Implementation plan (PR to follow)

1. `TrainingService` holds a single `RunManager` instance (replaces module-level `_RUNS`).
2. `trainer.sft`/`trainer.rlsl` handlers deserialize full `TrainerConfig` and call `RunManager.start_sft`/`start_rlsl`.
3. New handlers map 1:1 to `RunManager` methods (`runs.list/status/progress/stop`, `presets.list`, `datasets.list/preview`, `adapters.list/delete`).
4. `run_rlsl` drops the `method∈dpo|orpo|grpo` restriction (allow `rlsl`).
5. Update `tests/test_trainer_dispatcher.py` handler-key pin (currently 5 keys at line 38-44).
6. Add `fusion-trainer` to `pyproject.toml` optional deps so CI runs trainer tests (currently skipped via `_HAS_FUSION_TRAINER` guard).

## Why now

GUI panel work (fusion-studio side, separate issue) blocks on this RPC surface. fusion-trainer's RunManager is shipped + tested (16 tests green), so the upstream gap is purely on the agent-studio dispatch side.

## Related

- fusion-trainer RunManager: `fusion_trainer/run_manager.py`
- GUI contract: `fusion-trainer/architecture/fusion-trainer-gui.md`
